zenoh/tests/shm.rs

```diff
@@ -27,19 +27,23 @@ use std::{
 use zenoh::{
     qos::{CongestionControl, Reliability},
     shm::{
-        BlockOn, GarbageCollect, PosixShmProviderBackend, ShmProviderBuilder, POSIX_PROTOCOL_ID,
+        AllocAlignment, BlockOn, GarbageCollect, MemoryLayout, PosixShmProviderBackend,
+        ShmProviderBuilder, POSIX_PROTOCOL_ID,
     },
     Session, Wait,
 };
 use zenoh_core::ztimeout;
+use zenoh_shm::api::buffer::traits::OwnedShmBuf;
 
 const TIMEOUT: Duration = Duration::from_secs(60);
 const SLEEP: Duration = Duration::from_secs(1);
 
 const MSG_COUNT: usize = 1_00;
 const MSG_SIZE: [usize; 2] = [1_024, 100_000];
 
-async fn open_session_unicast(endpoints: &[&str]) -> (Session, Session) {
+async fn open_session_unicast<const NO_SHM_FOR_SECOND_PEER: bool>(
```

Comment:
Just out of curiosity. Is there any reason to use a const generic rather than a plain bool argument?